cli/flags.rs

```diff
@@ -2647,6 +2669,22 @@ mod tests {
     );
   }
 
+  #[test]
+  fn allow_env_allowlist() {
```

Comment:
Please add a test case for multiple keys and some test case that will error in validator for that values